### ISSUE TITLE
Multi-GPU Dataloader sampling

### ISSUE THREAD

**Issue by NigelNelson** [moojink/openvla-oft#51]
First, thank you for this great work and making it available to the larger robotics community!

I may be missing a detail in the [RLDSDataset](https://github.com/moojink/openvla-oft/blob/main/prismatic/vla/datasets/datasets.py#L94), but I believe there is a bug in the data sampling for multi-gpu training jobs using `torchrun`.

The dataloader is initialized in the [finetune.py](https://github.com/moojink/openvla-oft/blob/main/vla-scripts/finetune.py#L1005C18-L1005C28) script as follows:
```python
dataloader = DataLoader(
        train_dataset,
        batch_size=cfg.batch_size,
        sampler=None,
        collate_fn=collator,
        num_workers=0,  # Important: Set to 0 if using RLDS, which uses its own parallelism
    )
```
With no sampler specified, it will default to a [SequentialSampler](https://github.com/hughperkins/pytorch-pytorch/blob/master/torch/utils/data/dataloader.py#L373). With `torchrun` each GPU will have a copy of this dataloader and iterate over the same indices.

However, I do not see a seed being set during finetuning, so is the assumption that the [shuffling that occurs](https://github.com/moojink/openvla-oft/blob/213c5ff8d6e9673885d7e4dd6f2ce73b5d8a45db/prismatic/vla/datasets/rlds/dataset.py#L569) in the `make_interleaved_dataset()` function will inherently randomize the indices sampled across GPUs?

**Comment by yiyeChen** [comment]
I believe the reason is that RLDSDataset is an IterableDataset and DistributedSampler is not compatible with IterableDataset. So I think the current mechanism is what you described - each process randomly samples a different batch from the dataset due to different seeds, with possible duplication across different iterations. But I don't understand the motivation of using this IterableDataset + no Distributed Sampler design. I guess it does not matter as the paper quantifies the training complexity as the number of gradient steps instead of epochs.
